internal/server/httpapi/trigger.go

```diff
@@ -0,0 +1,398 @@
+					// as a Variable_Str here, and when we go to parse the variables
+					// later we do the proper string versus HCL check in variables.go
+					v.Value = &pb.Variable_Str{Str: value}
+				}
+
+				variableOverrides = append(variableOverrides, v)
+			}
+		}
+
+		var resp *pb.RunTriggerResponse
+		runTriggerReq := &pb.RunTriggerRequest{
+			Ref: &pb.Ref_Trigger{
+				Id: runTriggerId,
+			},
+			VariableOverrides: variableOverrides,
+		}
+
+		// attempt to make a grpc request to run trigger by id
+		if requireAuth {
+			resp, err = client.RunTrigger(ctx, runTriggerReq)
+		} else {
+			resp, err = client.NoAuthRunTrigger(ctx, runTriggerReq)
+		}
+		if err != nil {
+			log.Error("server failed to run trigger", "id", runTriggerId, "err", err)
+
+			if status.Code(err) == codes.PermissionDenied {
+				http.Error(w,
+					fmt.Sprintf("request not authorized to run trigger: %s", err),
+					http.StatusUnauthorized)
+			} else {
+				// improve http error code, which is more applicable for general queue failures?
+				http.Error(w,
+					fmt.Sprintf("server failed to run trigger: %s", err),
+					http.StatusPreconditionFailed)
+			}
+
+			return
+		}
+		if resp == nil {
+			http.Error(w,
+				fmt.Sprintf("server returned no job ids from run trigger %q", runTriggerId),
+				http.StatusInternalServerError)
+			return
+		}
+		jobIds := resp.JobIds
+
+		streamOutput := r.URL.Query().Get("stream")
+
+		// Attempt to stream output back, on request.
+		if streamOutput != "" {
+			if !requireAuth {
+				// We do not allow streaming job stream info if a no-auth token trigger was requested
+				log.Trace("server does not allow for streaming job stream output for no-token trigger URLs")
+				w.WriteHeader(http.StatusNoContent)
+				return
+			}
+
+			log.Debug("attempting to stream back queued job output from running trigger")
+
+			cn, ok := w.(http.CloseNotifier)
+			if !ok {
+				log.Error("failed to stream job output, could not create http.CloseNotifier")
+				http.NotFound(w, r)
+				return
+			}
+			flusher, ok := w.(http.Flusher)
+			if !ok {
+				log.Error("failed to stream job output, could not create http.Flusher")
+				http.NotFound(w, r)
+				return
+			}
+
+			// Send the initial headers saying we're gonna stream the response.
+			w.Header().Set("Transfer-Encoding", "chunked")
+			w.WriteHeader(http.StatusOK)
+			flusher.Flush()
+
+			enc := json.NewEncoder(w)
+
+			// NOTE(briancain): We skip every two jobs here because when we call RunTrigger
+			// via gRPC, it eventually queues the trigger jobs through on-demand runners, and
+			// queueJobMulti returns three jobs: StartTask, the job to be queued, and StopTask. People
+			// really only expect output from the job to be queued, so we only stream that back.
+
+			// For example, a trigger that queues 2 Waypoint operation jobs returns 6 total job ids:
+			// Job List: [ {0: StartTask, 1: WP Operation 1, 2: StopTask}, {3: StartTask, 4: WP Operation 2, 5: StopTask}, ... ]
```

Comment:
We're selecting Waypoint operations from this list. In reality, the list is flat and not grouped like this, but I choose to show them grouped this way because the surrounding job ids (i.e. Start and Stop task) are directly related to the job they wrap.

Reply:
Perhaps removing the brackets in the comment will be less confusing to the reader? I can see now how it might make them think the job ids come in some kind of grouped set rather than a flat slice.

Reply:
Yeah I would say remove the brackets IMO. I read that as a slice of structs, oddly enough.

Reply:
Good idea, got that updated now!